src/html.pre.js

```diff
@@ -20,6 +20,17 @@ function pre(context) {
   const { document } = context.content;
   const $ = jquery(document.defaultView);
 
+  /* workaround until cache purge is properly setup */
+  let queryString = '';
+  let separator = '?';
+  Object.keys(context.request.params).forEach((key) => {
+    queryString += `${separator}${key}=${context.request.params[key]}`;
+    separator = '&';
+  });
+  // eslint-disable-next-line no-param-reassign
+  context.request.queryString = queryString;
+  /* end of workaround */
+
```

Comment:
@kptdobe what about:

```
context.request.queryString = context.request.url.substring(context.request.path.length);
```
?

Reply:
Thanks. My "final" version of the code has changed a little bit and includes your proposed changed. It is now here: https://github.com/adobe/helix-pages/tree/esi-cache-killer-debugging for history (in case we still want to use it at one point).
I reverted this PR anyway.